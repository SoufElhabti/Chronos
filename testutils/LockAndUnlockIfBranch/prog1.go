package pkg

import (
	"math/rand"
	"sync"
)

func main() {
	mutex := sync.Mutex{}
	if rand.Int() > 0 {
		mutex.Lock()
	}
	a = 5
	if rand.Int() > 0 {
		mutex.Unlock()
	}
	a = 6
}
