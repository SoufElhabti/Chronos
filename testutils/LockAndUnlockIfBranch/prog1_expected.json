{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {
			"t01829642": 1829745
		}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"Pos": 1829679,
			"Value": 1829676,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 2,
			"Pos": 1829679,
			"Value": 0,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 3,
			"Pos": 0,
			"Value": 1829679,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 4,
			"Pos": 1829704,
			"Value": 0,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 5,
			"Pos": 1829704,
			"Value": 1829604,
			"OpKind": 1,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 5
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 6,
			"Pos": 1829725,
			"Value": 1829722,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 7,
			"Pos": 1829725,
			"Value": 0,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 8,
			"Pos": 0,
			"Value": 1829725,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 9,
			"Pos": 1829752,
			"Value": 0,
			"OpKind": 0,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 9
				},
				"LocksetJson": null
			}
		},
		{
			"ID": 10,
			"Pos": 1829752,
			"Value": 1829604,
			"OpKind": 1,
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 10
				},
				"LocksetJson": null
			}
		}
	]
}